From 48d29d3cb23bdee586eace3001390d5b240f12b5 Mon Sep 17 00:00:00 2001
From: Serge BARANDAO <serge.barandao-ext@st.com>
Date: Mon, 3 Jun 2024 15:35:28 +0200
Subject: [PATCH 605/841] feat(stm32mp21): enable crypto peripherals on
 STM32MP215F-DK board
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

enable SAES, HASH, PKA and RNG1 and RNG2 as they are needed during boot
process.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 310d9ac2db928b25d51f3c08a80f0236804d57c0 feat(stm32mp21): enable crypto peripherals on STM32MP215F-DK board]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: Ib7dc44314c2d56c12497653d812d10b5917839c1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/381540
Reviewed-by: Serge BARANDAO <serge.barandao-ext@st.com>
Tested-by: Serge BARANDAO <serge.barandao-ext@st.com>
Domain-Review: Serge BARANDAO <serge.barandao-ext@st.com>
---
 fdts/stm32mp215f-dk.dts | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index 8a33e2d40..344338d47 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -30,6 +30,10 @@
 	};
 };
 
+&hash1 {
+	status = "okay";
+};
+
 &i2c3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c3_pins_a>;
@@ -46,6 +50,22 @@
 	status = "okay";
 };
 
+&pka {
+	status = "okay";
+};
+
+&rng1 {
+	status = "okay";
+};
+
+&rng2 {
+	status = "okay";
+};
+
+&saes {
+	status = "okay";
+};
+
 &usart2 {
 	pinctrl-names = "default", "idle", "sleep";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

